core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestRemoveEmptyMergeWriterRuleSet.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import io.trino.spi.connector.CatalogHandle;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.sql.planner.Partitioning;
+import io.trino.sql.planner.PartitioningScheme;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.iterative.Rule;
+import io.trino.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.trino.sql.planner.iterative.rule.test.PlanBuilder;
+import io.trino.sql.planner.plan.Assignments;
+import io.trino.sql.planner.plan.ExchangeNode;
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.TableFinishNode;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.sql.planner.SystemPartitioningHandle.SINGLE_DISTRIBUTION;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.testing.TestingHandles.createTestCatalogHandle;
+
+public class TestRemoveEmptyMergeWriterRuleSet
+        extends BaseRuleTest
+{
+    private CatalogHandle catalogSupportingScaledWriters;
+    private SchemaTableName schemaTableName;
+
+    @BeforeClass
+    public void setup()
+    {
+        catalogSupportingScaledWriters = createTestCatalogHandle("catalog");
+        schemaTableName = new SchemaTableName("schema", "table");
+    }
+
+    @DataProvider
+    public static Object[][] testRemoveEmptyMergeRewriteParams()
+    {
+        return new Object[][] {
+                {RemoveEmptyMergeWriterRuleSet.removeEmptyMergeWriterRule(), false},
+                {RemoveEmptyMergeWriterRuleSet.removeEmptyMergeWriterWithExchangeRule(), true}
```

Comment:
Data provider values need to have nice toString, otherwise it's hard to tell which test invocation failed

<img width="1217" alt="image" src="https://user-images.githubusercontent.com/144328/229081966-c1adcf7b-fee4-467e-847c-a8433a1915af.png">

you can avoid this problem by extracting test logic into a common method

```
    @Test
    public void testRemoveEmptyMergeRewrite()
    {
        testRemoveEmptyMergeRewrite(removeEmptyMergeWriterRule(), false);
    }

    @Test
    public void testRemoveEmptyMergeRewriteWithExchange()
    {
        testRemoveEmptyMergeRewrite(removeEmptyMergeWriterWithExchangeRule(), true);
    }

    private void testRemoveEmptyMergeRewrite(Rule<TableFinishNode> rule, boolean planWithExchange)
```

Reply:
Done